### ISSUE TITLE
goog.forwardDeclare does not work with new type inference

### ISSUE THREAD

**Issue by nicks** [google/closure-compiler#1256]
Consider the following code:
```
var goog = {}
goog.forwardDeclare = function (x) {}
/** @const */ var x = {}
/** @constructor */ x.Foo = function () {}

goog.forwardDeclare('x.Foo2')

/** @param {x.Foo|x.Foo2} y */
function foo(y) {
}
foo(1)
```

If I run the compiler with old type inference, I get the warning
```
input0:11: WARNING - actual parameter 1 of foo does not match formal parameter
found   : number
required: (null|x.Foo|x.Foo2)
```

If I run the compiler with new type inference, I get the warning:
```
input0:8: WARNING - A union type that includes ? is equivalent to ?
/** @param {x.Foo|x.Foo2} y */
```

how can i get new type inference to process this code correctly?

for reference, here is the debugger link:
http://closure-compiler-debugger.appspot.com/#input0%3Dvar%2520goog%2520%253D%2520%257B%257D%250Agoog.forwardDeclare%2520%253D%2520function%2520(x)%2520%257B%257D%250A%252F**%2520%2540const%2520*%252F%2520var%2520x%2520%253D%2520%257B%257D%250A%252F**%2520%2540constructor%2520*%252F%2520x.Foo%2520%253D%2520function%2520()%2520%257B%257D%250A%250Agoog.forwardDeclare('x.Foo2')%250A%250A%252F**%2520%2540param%2520%257Bx.Foo%257Cx.Foo2%257D%2520y%2520*%252F%250Afunction%2520foo(y)%2520%257B%250A%257D%250Afoo(1)%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CHECK_TYPES_NEW_INFERENCE%3D1%26CLOSURE_PASS%3D1%26MISSING_PROPERTIES%3D1

**Comment by dimvar** [comment]
Should be fixed at commit 8297e5cbc811fdcedf50acce814ffe410cd7f40d. I'm guessing it's not part of the release yet and you're not using head?

**Comment by nicks** [comment]
cool, thanks
